pkg/generate/app/app.go

```diff
@@ -182,17 +182,34 @@ func (r *ImageRef) NameReference() string {
 }
 */
 
+// nameAndTag returns an image name with a tag. If the tag is blank, it will
+// use the default tag ("latest")
+func nameAndTag(name, tag string) string {
+	if len(tag) == 0 {
+		tag = imageapi.DefaultImageTag
+	}
+	return fmt.Sprintf("%s:%s", name, tag)
+}
+
 // ObjectReference returns an object reference from the image reference
 func (r *ImageRef) ObjectReference() *kapi.ObjectReference {
-	if r.Stream != nil {
+	switch {
+	case r.Stream != nil:
+		return &kapi.ObjectReference{
+			Kind:      "ImageStreamTag",
+			Name:      nameAndTag(r.Stream.Name, r.Tag),
+			Namespace: r.Stream.Namespace,
+		}
+	case r.AsImageStream:
```

Comment:
It can be pre-existing.... you can say 'new-app openshift/mystream~http://mysource'. We won't generate a proper reference to mystream if it's in a different namespace (openshift) than my current project